Resources/public/js/views/ez-usermenuview.js

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-usermenuview', function (Y) {
+    'use strict';
+    /**
+     * Provides the User Menu View class
+     *
+     * @module ez-usermenuview
+     */
+    Y.namespace('eZ');
+
+    var CLASS_HIDDEN = 'is-user-menu-hidden',
+        CLASS_USER_PROFILE = 'ez-user-profile',
+        SELECTOR_USER_PROFILE = '.' + CLASS_USER_PROFILE,
+        SELECTOR_USER_MENU = '.ez-user-menu';
+
+    /**
+     * The User Menu view
+     *
+     * @namespace eZ
+     * @class UserMenuView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.UserMenuView = Y.Base.create('userMenuView', Y.eZ.TemplateBasedView, [], {
+        events: {
+            '.ez-user-menu-item': {
+                'tap': '_toggleUserMenu'
+            },
+            '.ez-user-menu-item-logout': {
+                'tap': '_logOut'
+            }
+        },
+
+        initializer: function () {
+            this.on('*:toggleUserMenu', this._toggleUserMenu);
+        },
+
+        /**
+         * Renders the user menu view
+         *
+         * @method render
+         * @return {eZ.UserMenuView} the view itself
+         */
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template(this.getAttrs()));
+            container.on('clickoutside', this._hideMenuOnClickOutside, this);
+
+            this._renderMenuItems();
+
+            return this;
+        },
+
+        /**
+         * Tap event handler on the logout link
+         *
+         * @method _logOut
+         * @protected
+         * @param event {Object} tap event facade
+         */
+        _logOut: function (event) {
+            event.preventDefault();
+            this.fire('logOut', {
+                originalEvent: event
+            });
+        },
+
+        /**
+         * Toggles the visibility of a user menu.
+         *
+         * @method _toggleUserMenu
+         * @protected
+         */
+        _toggleUserMenu: function () {
+            var methodName = this.get('displayed') ? 'addClass' : 'removeClass',
+                userMenu = this.get('container').one(SELECTOR_USER_MENU);
+
+            userMenu[methodName](CLASS_HIDDEN);
+            this.set('displayed', !this.get('displayed'));
+        },
+
+        /**
+         * Checks if clicked outside the user profile container and hides a user menu if it is visible.
+         *
+         * @method _hideMenuOnClickOutside
+         * @protected
+         */
+        _hideMenuOnClickOutside: function (event) {
+            if (!event.target.hasClass(CLASS_USER_PROFILE) && event.target.ancestor(SELECTOR_USER_PROFILE) === null && this.get('displayed')) {
```

Comment:
ok this is happening because the `tap` event used in eZ.UserProfileView to show the menu is not prevented, so with:

```js
        _toggleUserMenu: function (e) {
            e.preventDefault();
            this.get('userMenuView').toggleDisplayed();
        }
```

you can simplified the condition.